### PR TITLE
Fix linter stdout/stderr assertions

### PR DESCRIPTION
**Pull request by ascandella**
I noticed while working on a new linter that my test was passing even though the stdout/stderr was incorrect.

This seems to be fallout from this refactor:
https://github.com/samuelmeuli/lint-action/commit/73e9574651632c461b95746ebb53b6f117d99fb6

I had to update the timezone date regex to allow my timezone as well.

I also updated the `gofmt` linter, since it sends its output to stdout. I also fixed the tests for it.

I updated the prettier test to remove the fix assertion on stdout because, on my machine, it outputs something like:

```
    Expected: ""
    Received: ".prettierrc.json 26ms
    file1.js 18ms
    file2.css 45ms
    package.json 6ms"
```

to stdout.

### CODE REVIEW

**Comment by ascandella** [comment]
The linter check doesn't seem to be checking out my branch properly (since it's on my fork, not on this repo):

```
  stderr: "error: pathspec 'fix-assertions' did not match any file(s) known to git\n"
```

File: test/linters/params/prettier.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ function getFixParams(dir) {
 		// Expected output of the linting function
 		cmdOutput: {
 			status: 0,
-			stdout: "",
+			// NOTE: no stdout because it includes timing output
```
Comment (samuelmeuli) [inline test/linters/params/prettier.js]: I'd suggest replacing this with a test for `stdoutParts` instead:

```js
	const stdoutFile1 = `file1.js`;
	const stdoutFile2 = `file2.css`;
	return {
		// Expected output of the linting function
		cmdOutput: {
			status: 0,
			stdoutParts: [stdoutFile1, stdoutFile2],
		},
```

**Comment by samuelmeuli** [comment]
The "Lint" workflow is currently failing because of #13, that's not an issue with your code :)

**Comment by samuelmeuli** [comment]
Thanks!

**Pull request merged**